shapeworks_cloud/core/rest.py

```diff
@@ -82,8 +82,15 @@ def get_serializer_class(self):
     def create(self, request, **kwargs):
         data = request.data
         data['dataset'] = models.Dataset.objects.get(id=data['dataset'])
+        try:
+            data['last_cached_analysis'] = models.CachedAnalysis.objects.get(
+                id=data['last_cached_analysis']
+            )
+        except models.CachedAnalysis.DoesNotExist:
```

Comment:
I think this needs to be `except (KeyError, models.CachedAnalysis.DoesNotExist):`, since `data['last_cached_analysis']` will throw a `KeyError` if it is unset.

Reply:
Nice catch. Changed in [dee1666](https://github.com/girder/shapeworks-cloud/pull/198/commits/dee16664fb3b1174d54a3617eac9866d40261933)